presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -172,6 +173,7 @@ void PrestoServer::run() {
   protocol::registerHiveConnectors();
   protocol::registerTpchConnector();
 
+  velox::process::addDefaultFatalSignalHandler();
```

Comment:
@pranjalssh There is a shut down signal handler https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/PrestoServer.cpp#L93 that does graceful shutdown when process receives a SIGTERM. 

Ideally, all signal handling should be inside the application, instead of the library. Should we keep the fatal handler in Presto and call velox when a SIGSEGV is received by the process? cc: @bikramSingh91

Reply:
Abandoning as this is needed to be called before folly init is called

Reply:
@amitkdutta the DefaultFatalSignalHandler is just a default one to showcase how you can access and print the thread local variable. Library does not need to handle this, you can write your own signal handler and access the thread local variable to write your own custom messages, etc